feat(curriculum): forum leaderboard project (#50241)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/644b736c83c5250181633899.md

---
id: 644b736c83c5250181633899
title: Step 65
challengeType: 0
dashedName: step-65
---

# --description--

The `hasOwnProperty()` method is used to check if a specific property exists in an object. It returns `true` if the object has the property, otherwise `false`. For example, consider an object `person` with a property `name`:

```js
const person = {
  name: 'John'
};

console.log(person.hasOwnProperty('name')); // Output: true
console.log(person.hasOwnProperty('age')); // Output: false
```

> You will create a `if` statement to check if the argument named `id` is available to the `categorieList` object. To do this task, you will need to use the `hasOwnProperty()` method.

Reviewer: ```suggestion
You will create an `if` statement to check if the argument named `id` is available to the `categoryList` object. To do this task, you will need to use the `hasOwnProperty()` method.
```

Author: fixed

# --hints--

> You should use the `hasOwnProperty()` method to check if the `id` argument is available to the `categorieList` object.

Reviewer: ```suggestion
You should use the `hasOwnProperty()` method to check if the `id` argument is available to the `categoryList` object.
```

```js

> assert(code.match(/if\s*\(\s*categorieList\s*\.hasOwnProperty\s*\(\s*id\s*\)\s*\)/));

Reviewer: ```suggestion
assert(code.match(/if\s*\(\s*categoryList\s*\.hasOwnProperty\s*\(\s*(?:'id'|id)\s*\)\s*\)/));
```

Modified the regex to accept both arguments: `id` and `'id'`.

Author: properties are not strings in javascript objects

  --peach: #f28373;
  --salmon-color: #f0aea9;
  --light-blue: #8bd9f6;
  --light-orange: #f8b172;
  --light-green: #93cb5b;
  --golden-yellow: #f1ba33;
  --gold: #f9aa23;
  --green: #6bca6b;
}

body {
  background-color: var(--main-bg-color);
}

.fcc-logo{
  width: 210px;
  margin: auto;
  display: block
}

nav {
  background-color: var(--dark-navy);
  padding: 10px 0;
}

.title{
  text-align: center;
  margin: 10px 0;
  color: var(--light-grey);
}

table{
  width: 80%;
  color: var(--dark-grey);
  margin: auto;
  table-layout: fixed;
  border-collapse: collapse;
}

th:first-child {
  width: 70%;
  text-align: start;
}

th{
  border-bottom: 2px solid var(--dark-grey);
  padding-bottom: 10px;
  font-size: 1.2rem;
}

td{
  border-bottom: 1px solid var(--dark-grey);
  padding: 20px 0px;
}

td:not(:first-child) {
  text-align: center;
}

```

```js
const FORUM_LATEST = "https://forum-proxy.freecodecamp.rocks/latest";
const FORUM_TOPIC_URL = "https://forum.freecodecamp.org/t/";
const FORUM_CATEGORY_URL = "https://forum.freecodecamp.org/c/";
const AVATAR_URL = "https://sea1.discourse-cdn.com/freecodecamp";

const postsContainer = document.getElementById('posts-container');

> const categorieList = {

Reviewer: ```suggestion
const categoryList = {
```

Author: fixed
